IOPool/Streamer/interface/FRDEventMessage.h

```diff
@@ -121,11 +145,17 @@ class FRDEventMsgView {
   uint32 adler32() const { return adler32_; }
   uint32 crc32c() const { return crc32c_; }
 
+  //can only be MC with output module flag
+  bool isRealData() const {
+    return !(flags_ & FRDEVENT_MASK_FROMOUTPUTMODULE) || (flags_ & FRDEVENT_MASK_ISREALDATA);
```

Comment:
why not simply
```suggestion
    return (flags_ & FRDEVENT_MASK_ISREALDATA);
```
?

Reply:
I'm wary of 0 being used as MC=true in production.
If someone updates the version in EvB to use v6 or higher, makes updates in CMSSW but forgets about this, we can end up having data flagged as MC in HLT..(if all values are left 0).

If you think it should be simpler, we can invert isRealData and store "isMC" as a flag.

Reply:
Ah, wait, I may have confused the bits...

So with this PR
  - `flags = 0x00` means that the flags are not saved (like in the old format) so it's real data
  - `flags = 0x01` means that the flags are saved, and it's real data
  - `flags = 0x11` means that the flags are saved, and it's MC
  - `flags = 0x10` should not be possible

?

I was thinking we could keep the existing behaviour unchanged:
  - `flags = 0x00` means real data (either implicitly like in the old format, or explicitly for real data)
  - `flags = 0x01` means MC
  - `flags = 0x10` is not used
  - `flags = 0x11` is not used


If somebody reads a v5 file with an old release, they always get "real data".
If somebody reads a v6 file with an old release, they get an error.
If somebody reads a v5 file with a new release, they always get "real data".
If somebody reads a v6 file with a new release, they get "real data" or "MC" according to the flags.

Did I miss any case that could be a concern ?

Reply:
:+1:

Reply:
I will use a bit more generic FRDEVENT_MASK_ISGENDATA flag name - because there is also use case for DaqFakeReader which isn't MC, but random/fake fed data.